src/bin/extensions.mli

```diff
@@ -0,0 +1,29 @@
+type t
+
+val pp : Format.formatter -> t -> unit
+
+val name : t -> string
+(** Returns the name of the extension. *)
+
+val load_typing_extension :
+  t -> Loop.State.t -> (Loop.State.t, [> `Msg of string]) result
+(** [load_typing_extension e] loads and returns the typing extension associated
+    with [e].
```

Comment:
This does not return the typing extension, but rather the state after loading the extension.